src/main.rs

```diff
@@ -97,85 +257,38 @@ fn main() {
     let username = username.expect("<username> argument");
     let password = password.expect("<password> argument");
 
-    let mut target = TargetJson {
-        major_version: MAJOR_VERSION,
-        minor_version: MINOR_VERSION,
-        x: 0,
-        y: 0,
-        image: String::new()
-    };
-    let mut width = 0;
-    let mut height = 0;
-    let mut target_image = Vec::new();
+    let init_config = matches.value_of("config").unwrap_or(TARGET_JSON_URL);
 
     let mut board = Vec::new();
     board.resize(1000*1000/2, 0u8);
 
+    let root = Rc::new(RefCell::new(Job::new(init_config.to_string())));
+
     loop {
-        let mut try_place_pixel = || -> Result<(), Box<Error>> {
-            let new_target: TargetJson = reqwest::get(TARGET_JSON_URL)?.json()?;
-            if new_target.major_version > MAJOR_VERSION {
-                println!("New major version is available. Must update!");
-                process::exit(1);
-            }
+        let mut queue = VecDeque::new();
+        queue.push_front(root.clone());
 
```

Comment:
I'm using a queue to get proper fallback ordering - first it should try all direct fallbacks, then all fallbacks of the first fallback, then all fallbacks of the second fallback, etc.